server/bin/www

```diff
@@ -34,23 +34,29 @@ const io = require("socket.io")(server);
 io.on("connection", (socket) => {
   socket.on("go-online", (id) => {
     if (!onlineUsers.includes(id)) {
-      onlineUsers.push(id);
+      onlineUsers.addUser(id, socket.id);
     }
     // send the user who just went online to everyone else who is already online
     socket.broadcast.emit("add-online-user", id);
   });
 
   socket.on("new-message", (data) => {
-    socket.broadcast.emit("new-message", {
+    let user = onlineUsers.getUser(data.recipientId);
```

Comment:
We can use const here - as a general rule of thumb, it's always best to use const over let when you can